app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,64 @@
+class OrdersController < ApplicationController
+  def index
+    @orders = Order.for_user(current_user)
+  end
+
+  def show
+    @order = Order.find(params[:id])
+  end
+
+  def no_cart
+    flash[:alert] = 'you have not put anything in the cart'
+    redirect_to root_path
+  end
+
+  def create
+    if current_user.present?
+      order = find_order
+      if order.blank?
+        params[:alert] = "This Order is not belongs to you"
```

Comment:
Wen, did you mean `flash[:alert]`? If so, why not just pass it as argument to the `redirect_to` bellow